tests/Doctrine/Tests/DBAL/Query/QueryBuilderTest.php

```diff
@@ -882,4 +882,73 @@ public function testJoinWithNonUniqueAliasThrowsException()
 
         $qb->getSQL();
     }
+    
+    /**
+     * this test fails while it should not
+     */
+    public function testDoubleFromWithJoin()
+    {
+        $qb = new QueryBuilder($this->conn);
+
+        $qb->select('a.id, b.id, c.id')
+            ->from('table_a', 'a')
+            ->from('table_b', 'b')
+            ->join('a', 'table_c', 'c', 'a.fk_c = c.id')
+            ->where('a.fk_b = b.id');
+
+        $this->assertSame('SELECT a.id, b.id, c.id FROM table_a a, table_b b INNER JOIN table_c c ON a.fk_c = c.id WHERE a.fk_b = b.id', $qb->getSQL());
```

Comment:
> Shouldn't it be an error from QueryBuilder not database?

The query builder is a stupid string assembler: it doesn't nor has to check the validity of the SQL.